netbox/netbox/utils.py

```diff
@@ -24,3 +25,14 @@ def _wrapper(cls):
         return cls
 
     return _wrapper
+
+
+def register_system_job():
```

Comment:
What do you think about passing the interval (and potentially other parameters) via the `register_system_job()` wrapper instead of defining them inside the class? They would need to be stored in the registry alongside a reference to the class itself, but it might allow for more flexible reuse of jobs. You could do something like:

```python
register_system_job(interval=HOURLY, do_big_stuff=False)(MyHousekeepingJob)
register_system_job(interval=DAILY, do_big_stuff=True)(MyHousekeepingJob)
```

Just a thought; I might also be overthinking this. Happy to merge as-is.

Reply:
I can see the potential in this approach. Would it be okay if the decorator is the only way to register a system job even for plugins instead of listing them in the `system_jobs` variable at the end of the file? `@register_model_view` uses the same approach, but other APIs like data source registration do not allow direct access to the decorator.

Reply:
I think personally I prefer the decorator approach but I don't have a strong opinion. It would probably make sense to encourage the direct use of the decorator for data backend registration in plugins as well. @arthanson @bctiemann @DanSheps do any of you have thoughts here?

Reply:
We discussed this today and opted to go with the decorator approach as it feels "cleaner" and offers more flexibility.